src/components/CDPCreateScreens/CDPCreateSelectCollateral.js

```diff
@@ -81,16 +81,23 @@ const CDPCreateSelectCollateral = ({ selectedIlk, proxyAddress, dispatch }) => {
   const [loading, setLoading] = useState(true);
   const balances = useWalletBalances();
 
+  async function checkAllowance(gemToken) {
+    const allowance = (await gemToken.allowance(
+      maker.currentAddress(),
+      proxyAddress
+    )).toNumber();
+
+    return allowance === MAX_UINT_BN.toNumber();
+  }
+
   useEffect(() => {
     (async () => {
       setLoading(true);
       try {
         const gemToken = maker.getToken(selectedIlk.currency.symbol);
         const hasAllowance =
           selectedIlk.currency.symbol === 'ETH' ||
-          (await gemToken.allowance(maker.currentAddress(), proxyAddress)).eq(
```

Comment:
@abylaw @levity 
for some reason, `.eq` is returning false when comparing the currency object of the allowance, and the MAX_UINT_BN (big number). Even when I convert the BigNumber to the same currency type, `.eq` is still false. I thought doing something like this would work with `.isEqual` (https://github.com/makerdao/currency/blob/master/src/Currency.js#L22)
```
const currency = gemToken._currency(MAX_UINT_BN);
const allowance = gemToken.allowance(maker.currentAddress(), proxyAddress);

currency.eq(allowance)); // returns false
currency.gt(allowance)); // returns true
```
The variables look identical in the console, converting them with `.toNumber()` results with equality. Any thoughts, or do you think this solution is ok for checking allowance?

Reply:
So I've noticed that the reason why allowance checks are failing are that any action on the CDP (withdraw, deposit...) makes the allowance some tiny amount less than the max. i.e. `MAX_UINT_BN - 1 !== MAX_UINT_BN`

I personally think this is fine way to approach it, given that as long as they have absolutely huge amount of allowance remaining, we're in no danger of a transaction failing.

It would be good to leave a comment about why this approach works though.

Reply:
> any action on the CDP (withdraw, deposit...) makes the allowance some tiny amount less than the max. i.e. MAX_UINT_BN - 1 !== MAX_UINT_BN

just a thought, if that tiny amount is consistent, we could make a BN const for that & compare.